main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -1189,13 +1188,12 @@ object GenExpression {
                                visitor: MethodVisitor,
                                jumpLabels: Map[Symbol.LabelSym, Label],
                                entryPoint: Label,
-                               op: UnaryOperator,
                                sop: SemanticOperator)(implicit root: Root, flix: Flix): Unit = {
     // Adding source line number for debugging
     addSourceLine(visitor, e.loc)
 
     compileExpression(e, visitor, currentClassType, jumpLabels, entryPoint)
-    op match {
+    SemanticOperatorOps.toUnaryOp(sop, e.loc) match {
```

Comment:
Oh no. This is not allowed :)

Reply:
The point is to make the backend indepedent of Unary/Binary OP

Reply:
Oh I just thought you wanted it removed from the AST node

Reply:
Ok, my bad. I did not explain well.

Think of it this way:

Unary/BinaryOp represents a "syntactic" operator, e.g +.
SemanticOp represents an actual operator, e.g. +_Int32 (addition specifically on Int32s).

We want to refactor the backend to not rely on the syntactic ones.

Reply:
So what you have done so far is fine, but we need to patmatch on the sop.

Reply:
Roger :+1: